src/main/kotlin/org/rust/lang/core/resolve/ItemResolution.kt

```diff
@@ -36,6 +36,13 @@ fun processItemOrEnumVariantDeclarations(
                 if (processAllWithSubst(item.variants, subst, processor)) return true
             }
         }
+        is RsImplItem -> {
+            val (item, subst) = (scope.typeReference?.skipParens() as? RsBaseType)
+                ?.path?.reference?.advancedDeepResolve() ?: return false
+            if (item is RsEnumItem) {
+                if (processAllWithSubst(item.variants, subst, processor)) return true
```

Comment:
Do I really need `subst` here? Maybe better to use `processAll` instead?

Reply:
Looks like yes: rustc shows an error in this case:
```rust
trait Foo {
    fn foo();
}

impl Foo for Option<i32> {
    fn foo() {
        let a: Option<u8> = Self::None;
    }
}
```
that means that rustc takes into account actual generic arguments of the enum